config/consensus.go

```diff
@@ -516,6 +516,18 @@ type ConsensusParams struct {
 	// used by agreement for Circulation, and updates the calculation of StateProofOnlineTotalWeight used
 	// by state proofs to use the same method (rather than excluding stake from the top N stakeholders as before).
 	ExcludeExpiredCirculation bool
+
+	// DynamicFilterTimeout specifies whether the timeout for the filter step
+	// should be determined dynamically, at run-time.
+	DynamicFilterTimeout bool
+
+	// DynamicFilterTimeoutLowerBound specifies a minimal duration that the
+	// filter timeout must meet.
+	DynamicFilterTimeoutLowerBound time.Duration
+
+	// DynamicFilterPayloadArriavalHistory specifies the number of past block arrivals
+	// that are measured to determine when to start pipelining the next block.
+	DynamicFilterPayloadArriavalHistory int
```

Comment:
This implicitly requires that the dynamic timeout be the 100%-th percentile of the arrival times over the last `DynamicFilterPayloadArriavalHistory` rounds.  This might be worth specifying in the comment.  (Seems like a reasonable policy to not choose something more aggressive like 95%-th percentile, etc.)